packages/analytics/amplify_analytics_pinpoint_dart/lib/src/exception/pinpoint_analytics_exception.dart

```diff
@@ -0,0 +1,63 @@
+import 'package:amplify_analytics_pinpoint_dart/src/sdk/pinpoint.dart';
+import 'package:amplify_core/amplify_core.dart';
+
+const _fileIssueMessage =
+    'This exception is not expected. Please try again. If the exception persists, please file an issue at https://github.com/aws-amplify/amplify-flutter/issues';
+
+/// {@template amplify_analytics_pinpoint_dart.pinpoint_exception}
+/// Represents exceptions that may be thrown calling Pinpoint Analytics plugin APIs.
+/// {@endtemplate}
+class PinpointException extends AnalyticsException {
+  /// {@macro amplify_analytics_pinpoint_dart.pinpoint_exception}
+  const PinpointException(
+    super.message, {
+    required String super.recoverySuggestion,
+    super.underlyingException,
+  });
+
+  /// Creates a [PinpointException] from a [BadRequestException].
+  factory PinpointException.fromBadRequestException(
+    BadRequestException exception,
+  ) {
+    return PinpointException(
+      'Error in request data sent to Pinpoint.',
+      recoverySuggestion: _fileIssueMessage,
+      underlyingException: exception,
+    );
+  }
+
+  /// Creates a [PinpointException] from a [NotFoundException].
+  factory PinpointException.fromNotFoundException(NotFoundException exception) {
+    return PinpointException(
+      'The Pinpoint resource was not found.',
+      recoverySuggestion:
+          'Please ensure Analytics is properly configured in your amplifyconfiguration.dart file.',
```

Comment:
Is this message helpful? When would this be expected to happen?

Reply:
The idea was to direct users to inspect their setup in case of this exception instead of throwing it within the generic AnalyticsException.  

In my previous tests I thought it was triggered on a amplifyconfig mistep but now only getting ForbiddenExceptions so we can remove this as I'm unsure when it's expected to happen if at all now.